crates/iota-data-ingestion-core/src/progress_store/file.rs

```diff
@@ -95,11 +98,30 @@ impl FileProgressStore {
 
     /// Writes the given data to the file, overwriting any existing content.
     async fn write_to_file(&mut self, data: impl AsRef<[u8]>) -> IngestionResult<()> {
-        // before writing seek to the start of the file
-        self.file.seek(SeekFrom::Start(0)).await?;
-        // clear the file content
-        self.file.set_len(0).await?;
-        Ok(self.file.write_all(data.as_ref()).await?)
+        let tmp_path = self.path.with_extension("tmp");
+
+        {
+            let mut tmp_file = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open(&tmp_path)
+                .await?;
+            tmp_file.write_all(data.as_ref()).await?;
+
```

Comment:
Should there be a `tmp_file.sync_all().await?;` after the `write_all` to ensure data is on disk?
https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_all

Isn't there a risk that rename() points to a file that was never flushed to disk if the system crashes.

Reply:
`sync_all()` was quite slow during testing, but as a tradeoff we could use the [sync_data()](https://docs.rs/tokio/latest/tokio/fs/struct.File.html#method.sync_data) since we only need the file content and not its metadata. 

what do you think?

Reply:
fixed: https://github.com/iotaledger/iota/pull/6918/commits/9fca597f1dc3ac4fa2069be048d4deaf4593eaf3